components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/ResponseBuilder.java

```diff
@@ -17,12 +17,44 @@
  */
 package org.wso2.carbon.identity.sso.saml.builders;
 
+import org.joda.time.DateTime;
 import org.opensaml.saml2.core.Response;
 import org.wso2.carbon.identity.base.IdentityException;
 import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
 
+import java.security.NoSuchAlgorithmException;
+
 public interface ResponseBuilder {
 
-    Response buildResponse(SAMLSSOAuthnReqDTO authnReqDTO, String sessionIndexId) throws IdentityException;
+    /**
+     * Build response using SAMLSSOAuthnReqDTO and session index.
+     *
+     * @param authnReqDTO    SAML sso authentication request DTO.
+     * @param sessionIndexId Session index ID.
+     * @return Built response object.
+     * @throws IdentityException
+     * @throws NoSuchAlgorithmException
+     * @deprecated Use {@link #buildResponse(SAMLSSOAuthnReqDTO, String, DateTime, String)} instead.
+     */
+    @Deprecated
+    Response buildResponse(SAMLSSOAuthnReqDTO authnReqDTO, String sessionIndexId)
+            throws IdentityException, NoSuchAlgorithmException;
```

Comment:
The deprecated method throws a new Exception which was not before.
This might break extensions already implemented. Ideally only the IdentityException should be thrown

Reply:
Fixed.